### ISSUE TITLE
Calendar entries missing but appears doubled on another account

### ISSUE THREAD

**Issue by CamZie** [nextcloud/calendar#1080]
### Steps to reproduce
Could not reproduce. Calendar entries that was shared to other users/groups suddenly disappeared

### Expected behaviour
The calendar entries should appear normally to every user

### Actual behaviour
Calendar entries suddenly disappeared but appears doubled on a newly created account:

- The first calendar has entries, which should be the correct one (which is somehow not appearing on other accounts, even on the original owner's account)
- The second calendar has no entries  (the current one that other accounts are seeing)

### Server configuration

**Operating system**: Debian 9.6

**Web server:** NGINX 1.10

**Database:** MariaDB 10.1.37

**PHP version:** PHP 7.0.33

**Server version:** 14.0.6

**Calendar version:** 1.6.4

**Updated from an older installed version or fresh install:** Updated from an older version

**Signing status (ownCloud/Nextcloud 9.0 and above):**

```
```

**List of activated apps:**

```
Enabled:
  - accessibility: 1.0.1
  - activity: 2.7.0
  - admin_notifications: 1.0.2
  - calendar: 1.6.4
  - cloud_federation_api: 0.0.1
  - comments: 1.4.0
  - dav: 1.6.1
  - federatedfilesharing: 1.4.0
  - federation: 1.4.0
  - files: 1.9.0
  - files_pdfviewer: 1.3.2
  - files_sharing: 1.6.2
  - files_texteditor: 2.6.0
  - files_trashbin: 1.4.1
  - files_versions: 1.7.1
  - files_videoplayer: 1.3.0
  - firstrunwizard: 2.3.0
  - gallery: 18.1.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.2.0
  - mail: 0.11.0
  - nextcloud_announcements: 1.3.0
  - notifications: 2.2.1
  - oauth2: 1.2.1
  - onlyoffice: 2.1.2
  - password_policy: 1.4.0
  - provisioning_api: 1.4.0
  - ransomware_protection: 1.2.0
  - serverinfo: 1.4.0
  - sharebymail: 1.4.0
  - spreed: 4.0.2
  - support: 1.0.0
  - survey_client: 1.2.0
  - systemtags: 1.4.0
  - tasks: 0.9.8
  - theming: 1.5.0
  - twofactor_backupcodes: 1.3.1
  - updatenotification: 1.4.1
  - workflowengine: 1.4.0
```

**Nextcloud configuration:**

```
<?php
$CONFIG = array (
  'passwordsalt' => '***REMOVED***',
  'secret' => '***REMOVED***',
  'trusted_domains' => 
  array (
    0 => '***REMOVED***',
  ),
  'datadirectory' => '***REMOVED***',
  'overwrite.cli.url' => 'https://***REMOVED***',
  'dbtype' => 'mysql',
  'version' => '14.0.6.0',
  'dbname' => '***REMOVED***',
  'dbhost' => 'localhost',
  'dbport' => '',
  'dbtableprefix' => 'oc_',
  'dbuser' => '***REMOVED***',
  'dbpassword' => '***REMOVED***',
  'installed' => true,
  'instanceid' => '***REMOVED***',
  'mail_smtpmode' => 'smtp',
  'log_rotate_size' => '10485760',
  'log_type' => 'errorlog',
  'loglevel' => '0',
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'trashbin_retention_obligation' => 'auto,90',
  'activity_expire_days' => 180,
  'logtimezone' => 'Europe/Zurich',
  'mail_from_address' => 'info',
  'mail_smtpauthtype' => 'LOGIN',
  'mail_domain' => '***REMOVED***',
  'onlyoffice' => 
  array (
    'jwt_secret' => '***REMOVED***',
    'jwt_header' => 'AuthorizationJwt',
  ),
  'mail_smtphost' => '***REMOVED***',
  'mail_smtpauth' => 1,
  'mail_smtpname' => '***REMOVED***',
  'mail_smtppassword' => '***REMOVED***',
  'maintenance' => false,
);
```

**Are you using encryption:** no

### Client configuration
**Browser:** Safari and Firefox

**Operating system:** Mac OS X

#### Log (not sure if relevant but this appears for every user)
```
2019/01/25 10:31:19 [error] 22454#22454: *5030091 FastCGI sent in stderr: "PHP message: [owncloud][webdav][0] {"Exception":"Sabre\\DAV\\Exception\\NotAuthenticated","Message":"No public access to this resource., No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured, No 'Authorization: Bearer' header found. Either the client didn't send one, or the server is mis-configured, No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured","Code":0,"Trace":[{"function":"beforeMethod","class":"Sabre\\DAV\\Auth\\Plugin","type":"->","args":[{"absoluteUrl":"https:\/\/***REMOVED***\/remote.php\/dav\/calendars\/***REMOVED***\/","__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]},{"file":"\/var\/www\/***REMOVED***\/3rdparty\/sabre\/event\/lib\/EventEmitterTrait.php","line":105,"function":"call_user_func_array","args":[[{"autoRequireLogin":true,"__class__":"Sabre\\DAV\\Auth\\Plugin"},"beforeMethod"],[{"absoluteUrl":"https:\/\/***REMOVED***\/remote.php\/" while reading response header from upstream, client: ***REMOVED***, server: ***REMOVED***, request: "PROPFIND /remote.php/dav/calendars/***REMOVED***/ HTTP/1.1", upstream: "fastcgi://unix:/var/run/***REMOVED***.ch.sock:", host: "***REMOVED***"
```
#### Browser log
```
JOMIGRATE: Migrate is installed, version 1.4.0
"Deprecation warning: use moment.updateLocale(localeName, config) to change an existing locale. moment.defineLocale(localeName, config) should only be used for creating a new locale See http://momentjs.com/guides/#/warnings/define-locale/ for more info."
Attempting to call a FullCalendar methon on an element with no calendar.
TypeError: xml is null [Weitere_Informationen]
```

**Comment by georgehrke** [comment]
Can you tell whether this was related to any update?

Do you remember having troubles with an update?

Do you use any user-backend like LDAP or IMAP? If yes, did you change the configuration of the user backend?

**Comment by CamZie** [comment]
@georgehrke Please find below answers to your questions:

Although I cannot tell with certainty, the problem seems to have appeared a few days after an update from NC version 13.0.6 to 14.0.6. No problem occurred during the upgrade.

No LDAP or IMAP back-end.

I checked the oc_calendars and saw several calendars with the same name for user1 and user2 (see example below).

Since the calendar in the example below was shared by user1 to user2, does it make sense that it has its own entries in oc_calendars? I think this is why user2 sees this calendar twice in their client.

I also checked the oc_calendarobjects and found no entries for user1 (original owner of the calendar), where there should have been around 200 entries.

To sum up, the original shared calendar of user1 is now empty and its original content only exists under the duplicate calendar of user2. The latter is not being synchronised with any other calendars.

```
           id: 69
 principaluri: principals/users/user1
  displayname: mycalendar
          uri: mycalendar
    synctoken: 179
  description: NULL
calendarorder: 11
calendarcolor: #74e7d2
     timezone: BEGIN:VCALENDAR

           id: 79
 principaluri: principals/users/user2
  displayname: mycalendar (user1)
          uri: A6764F1B-1B94-44E4-832E-BFA08C136453
    synctoken: 31
  description: NULL
calendarorder: 38
calendarcolor: #74E7D2FF
     timezone: BEGIN:VCALENDAR
```

**Comment by georgehrke** [comment]
@tcitworld Have you seen issues like this before?

I tried plenty of things, but I was never able to reproduce it.
And to my knowledge this is the only bug report we have gotten about this weird behaviour.

**Comment by tcitworld** [comment]
I don't think I heard anything like this, no.
There could be a lot of areas where it could have gone wrong (calendar, session, sharing, updating…) so let's close this for now.
